### DISCUSSION TITLE
Class package limit

### DISCUSSION THREAD

**Discussion by kuimovvg** [seaside1/jrule#169] [Q&A]
My question is, why is there a restriction that the class must be in org.openhab.automation.jrule.rules.user.* when classes are loaded from jar ?

**Comment by querdenker2k** [comment]
Why not?
Convention is a good idea. The whole openhab core packages are named by convention. JRule does this as well.

**Comment by seaside1** [comment]
Yes for rules I tend to agree that it's good with a convention. If you have a lot of utility functionality etc, you can place it in a separate jar-file with a different package structure and use that as a dependency for the rules.

**Reply by kuimovvg** [reply]
What is the convention here?
Let's imagine you are some kind of company with a domain, let it be test.com. You make some kind of product with some rules. Why would you write rules in packages "org.openhab.automation.jrule.rules.user". If you think that it should be like this by default, can you put such a setting in "$JRULE_ROOT/jrule.conf" with the default "org.openhab.automation.jrule.rules.user" ?

**Reply by seaside1** [reply]
If that was the case, would it not be possible to have the rules under the org.openhab..... package? Being that it utilizes openHab for executing those rules. I guess this addon was never designed with the mindset of being a tool for a company. openHAB is targeting "Home automation" and therefor this addon as well within the same context. 

I have never seen this request either. This after all opensource if you think it's a good idea to support rules under a different package you could always submit a PR to introduce the functionality.

**Reply by kuimovvg** [reply]
#170

**Comment by seaside1** [comment]
Merged
